evaluation.py

```diff
@@ -237,7 +237,97 @@ def predict_keypoints(self,generator, location):
                 image_ids.append(metadata['src_set_image_id'])
                 j+=1
         self.write_to_json_file(list_of_predictions, location)
-        return image_ids
+        return image_ids, list_of_predictions
+
+    def pck_eval(self, generator, location, annotation_file):
+        _, list_of_predictions = self.predict_keypoints(generator, location)
+        f = open(annotation_file,)
+        data = json.load(f)
+
+        dist_list = []
+        correct_keypoints = {
+            "nose": 0,
+            "left_eye": 0,
+            "right_eye": 0,
+            "left_ear": 0,
+            "right_ear": 0,
+            "left_shoulder": 0,
+            "right_shoulder": 0,
+            "left_elbow": 0,
+            "right_elbow": 0,
+            "left_wrist": 0,
+            "right_wrist": 0,
+            "left_hip": 0,
+            "right_hip": 0,
+            "left_knee": 0,
+            "right_knee": 0,
+            "left_ankle": 0,
+            "right_ankle": 0
```

Comment:
would bools make more sense here? `False`?

Reply:
Never mind, I understand how this is used now. This is across the entire set, right?

Reply:
Yeah this is keeping track of how many times each keypoint is "correct" and then it divides by the number of predictions were looking at!